src/InputWithLabel.jsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { useRef, useEffect } from "react";
+
+const InputWithLabel = ({ id, children, value, onInputChange }) => {
+    const inputRef = useRef();
+
+    useEffect(() => {
+        inputRef.current.focus()
+    }, [])
+
+    return (
+        <React.Fragment>
+            <label>{children}</label>
+            <input id={id} type='text' name={children} value={value} onChange={onInputChange} ref={inputRef} />
```

Comment:
The input can be improved for better accessibility and structure - I suggest formatting in the following way: 
```
        <React.Fragment>
            <label>{children}</label>
            <input
                id={id}
                type='text'
                name={children}
                value={value}
                onChange={onInputChange}
                ref={inputRef}
            />
            <label htmlFor={id}></label>
        </React.Fragment>
```